src/arizona_parse_transform.erl

```diff
@@ -96,7 +181,62 @@ templates that can be optimized at compile time.
     CompilerOptions :: compile_options(),
     AbstractSyntaxTrees1 :: [erl_parse:abstract_form()].
 parse_transform(AbstractSyntaxTrees, CompilerOptions) ->
-    parse_transform_with_depth(AbstractSyntaxTrees, CompilerOptions, 0).
+    ModuleName = extract_module_name(AbstractSyntaxTrees),
+    ArizonaFunctions = extract_arizona_functions(AbstractSyntaxTrees),
+
+    %% Selective processing: only process modules with explicit arizona_parse_transform attributes
+    case ArizonaFunctions of
+        [] ->
+            %% No Arizona functions declared - cause compilation error with proper module info
+            error(no_arizona_parse_transform_attribute, none, error_info(ModuleName));
+        _ ->
+            %% Validate that declared functions exist and are exported
+            validate_arizona_functions(AbstractSyntaxTrees, ArizonaFunctions, ModuleName),
+            %% Process declared Arizona functions normally using existing logic
+            parse_transform_with_depth(AbstractSyntaxTrees, CompilerOptions, 0)
+    end.
+
+%% Validate that declared Arizona functions exist and are exported
+-spec validate_arizona_functions(AbstractSyntaxTrees, ArizonaFunctions, ModuleName) -> ok when
+    AbstractSyntaxTrees :: [erl_parse:abstract_form()],
+    ArizonaFunctions :: [function_spec()],
+    ModuleName :: atom().
+validate_arizona_functions(AbstractSyntaxTrees, ArizonaFunctions, ModuleName) ->
+    %% Extract exported functions from module
+    ExportedFunctions = extract_exported_functions(AbstractSyntaxTrees),
+    %% Extract defined functions from module
+    DefinedFunctions = extract_defined_functions(AbstractSyntaxTrees),
+
+    %% Check each declared Arizona function
+    lists:foreach(
+        fun({FuncName, Arity}) ->
+            %% Check if function is defined
+            case lists:member({FuncName, Arity}, DefinedFunctions) of
+                false ->
+                    error(
+                        arizona_function_not_defined,
+                        none,
+                        error_info({ModuleName, FuncName, Arity, "Function not defined in module"})
+                    );
+                true ->
+                    %% Check if function is exported
+                    case lists:member({FuncName, Arity}, ExportedFunctions) of
```

Comment:
Maybe better to have this as a separate function.

Reply:
https://github.com/arizona-framework/arizona/pull/329/commits/b5aac3616abaf7d04dfc068d6f1c8dc1db4dcfb3